packages/mobile/src/screens/favorites-screen/AlbumsTab.tsx

```diff
@@ -27,10 +27,10 @@ export const AlbumsTab = () => {
   const [filterValue, setFilterValue] = useState('')
   const isReachable = useSelector(getIsReachable)
   const isOfflineModeEnabled = useIsOfflineModeEnabled()
-  const offlineDownloadStatus = useSelector(getOfflineDownloadStatus)
   const userAlbums = useProxySelector(
-    (state: CommonState) =>
-      getAccountCollections(state, filterValue).filter((collection) => {
+    (state: AppState) => {
+      const offlineDownloadStatus = getOfflineDownloadStatus(state)
```

Comment:
Ah, nice. So this rolls it all into one dependency?

How does this work if the status changes? I assume the hook runs but the component only rerenders if the result is different?

Reply:
Exactly! Any time the store state changes _all_ selectors in useSelectors are rerun and the result compared by default with reference equality

useProxySelector does this slightly more efficiently by tracking which parts of the state got accessed and then only rerendering if those parts changed

Reply:
Perfect. That's super cool. So we should be combining our complex selectors whenever possible.

Reply:
It's definitely a tradeoff, running a more complex computation every time the state changes vs rerendering a component more often. Usually rendering react components is wayyy more expensive though. Also there is some overhead to useProxySelector but I'm not sure how big